File: poker_game.js
function log(msg){
	console.log(msg);
}

function sum(arr) {
    var s = 0;
	for(i in arr){
		s += arr[i]
	}
	return s;
}

function pickPoker(data, user){
	select_row = Math.floor(Math.random() * data.length);
	pick_num = Math.ceil(Math.random() * data[select_row]);
	data[select_row] -= pick_num;
	if(data[select_row] == 0){
		data.splice(select_row, 1);
	}
	log(user + ' pick ' + pick_num + ' poker from row ' + (select_row + 1) + ", left pokers " + data);
	if(sum(data) == 0){
		return true;
	}else{
		return false;
	}
}

function FivePokerGame(){
	var gameData = [3, 5, 7];
	var users = ['User One', 'User Two'];
	var operator = Math.round(Math.random());
	log('=======> Game start, '+ users[operator] + ' first!');
	while(true){
		rlt = pickPoker(gameData, users[operator]);
		if(rlt == true){
			log('=======> Game over, ' + users[operator] + ' lose!');
			break;
		}
		operator = (operator + 1) % 2;
	}
	
}

FivePokerGame()

File: poker_game.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-
# @time:  14:18
# @Author: Alim
# @File: poker_game.py
# @Software: PyCharm
import random


def pick_poker(data, user):
    select_row = random.choice([key for key in data if data[key] != 0])
    # print(f'select row:{select_row}')
    pick_num = random.randint(1, data[select_row])
    data[select_row] -= pick_num
    print(f'{user} pick {pick_num} poker from row {select_row}, left poker {list(data.values())}')
    if sum([data[key] for key in data]) == 0:
        return True
    else:
        return False


def poker_game(data, users):
    operator = random.randint(0, len(users) - 1)
    print(f'=======> Poker game start, {users[operator]} first!')

    while True:
        rlt = pick_poker(data, users[operator])
        if rlt:
            print(f'=======> Game over, {users[operator]} lose')
            break
        operator = (operator + 1) % 2


if __name__ == '__main__':
    users = ['USER_One', 'USER_Two']
    game_data = {
        1: 3,
        2: 5,
        3: 7
    }

    poker_game(game_data, users)
